### PR TITLE
GEODE-5579: Add -PforceTest property to :test tasks

### PR DESCRIPTION
**Pull request by PurelyApplied**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [n/a] Have you written or updated unit tests to verify your changes?

- [n/a] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by PurelyApplied** [comment]
I noticed significant duplication in `test.gradle` with respect to the other test targets and have streamlined and unified that configuration.  If that seems beyond the scope of this ticket, I'm happy revert that last commit and make it its own ticket / let it be subsumed in the future work to remove `FlakyTest` as a whole.

**Comment by pivotal-jbarrett** [comment]
@PurelyApplied please remove the last commit or add a block to add the flaky exclusion back. This change will result in the remaining flaky tests being executed in the CI.

File: gradle/test.gradle [hunk 1]
```diff
@@ -144,60 +145,6 @@ subprojects {
     outputs.upToDateWhen { false }
   }
 
-  distributedTest {
-    useJUnit {
-      if (project.hasProperty("testCategory")) {
-        includeCategories += project.testCategory
-      }
-      excludeCategories 'org.apache.geode.test.junit.categories.FlakyTest'
```
Comment (rhoughton-pivot) [inline gradle/test.gradle]: This is the exclusion to re-insert in the configure() closure. I believe you can do it externally as well only for distributed and performance, if you don't want to if-check the names inside the configure() loop

**Comment by PurelyApplied** [comment]
I hadn't noticed that we got rid of the `FlakyTest` box in our pipeline.  That's awesome!  But yes, that does mean those tests aren't getting run elsewhere.

**Comment by pivotal-jbarrett** [comment]
@PurelyApplied the changes to the submodules seems to be negatively impacting the integration tests. Let's decouple those changes from this issue please.

**Comment by PurelyApplied** [comment]
@pivotal-jbarrett The integration failure appears to be due to a bad commit on / test-merge against `develop`.  See https://concourse.apachegeode-ci.info/teams/main/pipelines/develop/jobs/IntegrationTest/builds/257.

**Pull request merged**